fleet/fleet.go

```diff
@@ -0,0 +1,220 @@
+// Package fleet implements a fleet client providing basic operations against a
+// fleet endpoint through fleet's HTTP API. Higher level scheduling and
+// management should be built on top of that.
+package fleet
+
+import (
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/coreos/fleet/client"
+	"github.com/coreos/fleet/schema"
+	"github.com/coreos/fleet/unit"
+)
+
+type Config struct {
+	Client   *http.Client
+	Endpoint url.URL
+}
+
+func DefaultConfig() Config {
+	URL, err := url.Parse("file:///var/run/fleet.sock")
+	if err != nil {
+		panic(err)
+	}
+
+	newConfig := Config{
+		Client:   http.DefaultClient,
+		Endpoint: *URL,
+	}
+
+	return newConfig
+}
+
+// MachineStatus represents a unit's status scheduled on a certain machine.
+type MachineStatus struct {
+	// ID represents the machines fleet agent ID where the related unit is
+	// running on.
+	ID string
+
+	// IP represents the machines IP where the related unit is running on.
+	IP net.IP
+
+	// SystemdActive represents the unit's systemd active state.
+	SystemdActive string
+}
+
+// UnitStatus represents the status of a unit.
+type UnitStatus struct {
+	// Current represents the current status within the fleet cluster.
+	Current string
+
+	// Desired represents the desired status within the fleet cluster.
+	Desired string
+
+	// Machine represents the status within a machine. For normal units that are
+	// scheduled on only one machine there will be one MachineStatus returned.
+	// For global units that are scheduled on multiple machines there will be
+	// multiple MachineStatus returned. If a unit is not yet scheduled to any
+	// machine, this will be empty.
+	Machine []MachineStatus
+}
+
+// Fleet defines the interface a fleet client needs to implement to provide
+// basic operations against a fleet endpoint.
+type Fleet interface {
+	// Submit schedules a unit on the configured fleet cluster. This is done by
+	// setting the unit's target state to loaded.
+	Submit(name, content string) error
+
+	// Start starts a unit on the configured fleet cluster. This is done by
+	// setting the unit's target state to launched.
+	Start(name string) error
+
+	// Stop stops a unit on the configured fleet cluster. This is done by
+	// setting the unit's target state to loaded.
+	Stop(name string) error
+
+	// Destroy delets a unit on the configured fleet cluster. This is done by
+	// setting the unit's target state to inactive.
+	Destroy(name string) error
+
+	// GetStatus fetches the current status of a unit. If there cannot be any
+	// unit found, an error that you can identify using IsUnitNotFound is
```

Comment:
Done

On Fri, Feb 12, 2016 at 4:26 PM, Joseph Salisbury <notifications@github.com>
wrote:

> In fleet/fleet.go
> <https://github.com/giantswarm/formica/pull/13#discussion_r52754379>:
>
> > +	Submit(name, content string) error
> > +
> > +	// Start starts a unit on the configured fleet cluster. This is done by
> > +	// setting the unit's target state to launched.
> > +	Start(name string) error
> > +
> > +	// Stop stops a unit on the configured fleet cluster. This is done by
> > +	// setting the unit's target state to loaded.
> > +	Stop(name string) error
> > +
> > +	// Destroy delets a unit on the configured fleet cluster. This is done by
> > +	// setting the unit's target state to inactive.
> > +	Destroy(name string) error
> > +
> > +	// GetStatus fetches the current status of a unit. If there cannot be any
> > +	// unit found, an error that you can identify using IsUnitNotFound is
>
> This should read more like "if the unit cannot be found"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/giantswarm/formica/pull/13/files#r52754379>.
>



-- 
--
Tim Schindler
Developer @ Giant Swarm GmbH
tim@giantswarm.io
Web: http://giantswarm.io
Twitter: http://twitter.com/giantswarm
Handelsregister: Amtsgericht Köln, HRB 81484 · Sitz der Gesellschaft: Köln
· Geschäftsführer: Timo Derstappen, Henning Lange, Oliver Thylmann
The latest from Giant Swarm (@giantswarm). Simple Service Orchestration,
built for developers. Cologne, Germany